### PR TITLE
openfst: add python option and a description

### PR DESCRIPTION
**Pull request by Moisan**
Following issue #2920, this PR adds the option to include the python bindings. However I couldn't get `audit` to pass:

```
* python modules have explicit framework links
These python extension modules were linked directly to a Python
framework binary. They should be linked with -undefined dynamic_lookup
instead of -lpython or -framework Python.
  /usr/local/Cellar/openfst/1.5.0/lib/python2.7/site-packages/fst.so
```

How can I fix this?

### CODE REVIEW

**Comment by Moisan** [comment]
Also, to get the python module to work I had to run 
```
export PYTHONPATH=/usr/local/lib/python2.7/site-packages:$PYTHONPATH
``` 
Maybe it should be added to the formula caveat.

File: openfst.rb [hunk 1]
```diff
@@ -13,21 +12,31 @@ class Openfst < Formula
   end
 
   needs :cxx11
+  depends_on :python => :optional
```
Comment (dpo) [inline openfst.rb]: `unless OS.mac? && MacOS.version > :snow_leopard`

**Comment by dpo** [comment]
It seems your branch has conflicts. Would you mind rebasing against `master`?

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by Moisan** [comment]
@BrewTestBot test this please

**Comment by Moisan** [comment]
Jenkins seems to have issues unrelated with this formula (Failed to read test report file).

**Comment by dpo** [comment]
Sorry for the long delay. That's an intermittent Jenkins hiccup. But your formula needs a revision number.

**Comment by Moisan** [comment]
@BrewTestBot test this please

File: openfst.rb [hunk 2]
```diff
@@ -11,20 +13,30 @@ class Openfst < Formula
   end
 
   needs :cxx11
+  depends_on :python => :optional unless OS.mac? && MacOS.version > :snow_leopard
```
Comment (dpo) [inline openfst.rb]: Please convert this to an `if ... <= :snow_leopard` and squash your commits together. Thanks!

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by Moisan** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
@BrewTestBot test this please

**Comment by dpo** [comment]
Sorry (again) for the long delay. It seems your branch now has conflicts. Would you mind rebasing?

**Comment by iMichka** [comment]
@Moisan Could you please rebase this? We deliver Openfst 1.6.1 now. I can help with the audit failure, let's see what the bots are saying first.

**Comment by iMichka** [comment]
I gave this a try, and the problem is still there with the latest version. We need to open an issue upstream, I will do this during the weekend.

> These python extension modules were linked directly to a Python
framework binary. They should be linked with -undefined dynamic_lookup
instead of -lpython or -framework Python.
  /usr/local/opt/openfst/lib/python2.7/site-packages/pywrapfst.so

**Comment by iMichka** [comment]
Issue opened upstream: http://www.openfst.org/twiki/bin/view/Forum/FstForum#Do_not_link_against_python_libra

**Pull request closed without merging**